pkg/sql/rowenc/index_fetch.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package rowenc
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+	"github.com/cockroachdb/cockroach/pkg/util/buildutil"
+	"github.com/cockroachdb/errors"
+)
+
+// InitIndexFetchSpec fills in an IndexFetchSpec for the given index and
+// provided fetch columns. All the fields are reinitialized; the slices are
+// reused if they have enough capacity.
+//
+// The fetch columns are assumed to be available in the index. If the index is
+// inverted and we fetch the inverted key, the corresponding Column contains the
+// inverted column type.
+func InitIndexFetchSpec(
+	s *descpb.IndexFetchSpec,
```

Comment:
I get why you're doing this, but would it be better to make this a method on catalog.TableDescriptor and cache them on the leased descriptor wrapper objects? Seems like this thing is immutable and avoiding this re-computation and allocations seems good. As of https://github.com/cockroachdb/cockroach/pull/74722 we generally get a handle to the leased descriptor